automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/Notebook.scala

```diff
@@ -75,12 +74,34 @@ object Notebook extends RestClient with LazyLogging {
     val path = contentsPath(googleProject, clusterName, contentPath) + (if(includeContent) "?content=1" else "")
     logger.info(s"Get notebook contents: GET /$path")
     val cookie = Cookie(HttpCookiePair("LeoToken", token.value))
+
     handleContentItemResponse(parseResponse(getRequest(url + path, httpHeaders = List(cookie))))
   }
 
+  def getNotebookItem(googleProject: GoogleProject, clusterName: ClusterName, contentPath: String, includeContent: Boolean = true)(implicit token: AuthToken): NotebookContentItem = {
+    val path = contentsPath(googleProject, clusterName, contentPath) + (if(includeContent) "?content=1" else "")
+    logger.info(s"Get notebook contents: GET /$path")
+    val cookie = Cookie(HttpCookiePair("LeoToken", token.value))
+
+    handleNotebookContentResponse(parseResponse(getRequest(url + path, httpHeaders = List(cookie))))
+  }
+
   def setCookie(googleProject: GoogleProject, clusterName: ClusterName)(implicit token: AuthToken, webDriver: WebDriver): String = {
     val path = notebooksBasePath(googleProject, clusterName) + "/setCookie"
     logger.info(s"Set cookie: GET /$path")
     parseResponse(getRequest(url + path, httpHeaders = List(Authorization(OAuth2BearerToken(token.value)))))
   }
+
+  class NotebookMode()
```

Comment:
Use sealed trait

```
sealed trait NotebookMode extends Product with Serializable {
   def asString: String
}
object NotebookMode {
    final case object SafeMode extends NotebookMode {
    def asString: String = "playground"
   }

    final case object EditMode extends NotebookMode {
     def asString: String = "edit"
   }

   val stringToMode: Map[String, NotebookMode] = List(SafeMode, EditMode).map(x => x.asString -> x).toMap
}
```

Reply:
updated to sealed trait, but had to implement stringToMode slightly differently, as the string we are converting comes from the UI and lacks type safety